libraries/Microsoft.Bot.Builder/HealthCheck.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder
+{
+    internal class HealthCheck
+    {
+        public static HealthCheckResponse CreateHealthCheckResponse(IConnectorClient connector)
+        {
+            // A derived class may override this, however, the default is that the bot is healthy given we have got to here.
+            var healthResults = new HealthResults { Success = true };
+
+            if (connector != null)
```

Comment:
I'm confused over this. Should we throw if the connector is null? The behavior here isn't as obvious to me as it should be.

Reply:
If there is no connector the bot can't make callbacks - so there is no meaningful Authorization and no UserAgent - but in other respects if you be healthy.

This code sits in ActivityHandler, its not explicitly coupled to the BotFrameworkAdapter - could could imagine other adapters (without connectors) invoke this code.

For example a Skill that just supports expectReplies doesn't need the connector.